### ISSUE TITLE
bug: no Pending State for transactions with nonce too high

### ISSUE THREAD

**Issue by Eikix** [kkrt-labs/kakarot-rpc#984]
# Bug Report

**Kakarot RPC version:**

v0.6.2

### TL;DR
It is "surprising" for SDKs and tools that they can't send many transactions in any order with various nonces, as Kakarot does not have a "pending mempool" where txs await their nonce "too high" to become valid. Instead, Kakarot refuses nonce too highs at the mempool level.

Probable fix: to be discussed within the team

**Current behavior:**

Transactions that reach the RPC with a Nonce too high will fail asynchronously due to Starknet feeder gateway refusing txs with nonce too high. Leaving user without feedback regarding their tx and why it failed.

**Expected behavior:**

On Ethereum, each transaction from an account is assigned a sequential nonce. The nonce is a counter that represents the number of transactions sent from the account. It's crucial for preventing transaction replay attacks and ensuring transactions are processed in order.

When a transaction with a too high nonce reaches an Ethereum node's mempool (the waiting area for transactions before they are included in a block), the transaction will not be immediately executed or included in a block. This is because transactions from the same account must be processed in strict nonce order. Here's what typically happens:

### Pending State:

 The transaction with the too high nonce is considered "pending" and remains in the mempool. It waits for all transactions with lower nonces from the same account to be executed first. This ensures the sequential order of transactions.

### Queuing: 

Nodes may keep the transaction in a queue (also part of the mempool) specifically for transactions that cannot yet be processed due to nonce gaps. This queue helps nodes manage transactions that are out of order but expected to be valid eventually.
### Stuck Transactions:

 If transactions with the necessary nonce values to fill the gap do not appear, the high nonce transaction remains stuck in the mempool. It can stay there until it's dropped due to a timeout (nodes do not keep transactions in their mempool indefinitely) or until the nonce gap is filled.

### Manual Intervention: 

Users may notice that their transaction is stuck due to a too high nonce. They can intervene by either sending transactions with the missing nonce values or, in some cases, sending a zero-value transaction with a higher gas price to themselves to fill the nonce gap.
### Dropping:

 Eventually, if a transaction remains in the mempool without being able to be processed (due to nonce issues or otherwise), nodes may drop it to free up space. This dropping threshold varies between different Ethereum clients and their configurations.

### Rebroadcasting: 

Users can rebroadcast a dropped transaction if they still want it to be executed, provided they ensure that the nonce sequence is correct this time.
In summary, a transaction with a too high nonce does not disrupt the network but requires either the missing transactions to be broadcast or some form of manual intervention to proceed.

**Steps to reproduce:**

- git clone git@github.com:bajpai244/on-chain-whitelist.git
- git checkout feat/add-scripts

Fill .env
```
PRIVATE_KEY=
SEPOLIA_RPC_URL=https://sepolia-rpc.kakarot.org
```
then, run:
`make deploy`

Then look into the RPC logs, you will see that one of the two transactions failed, and upon querying the SN feeder gateway, it was refused because of Nonce error

**Comment by Eikix** [comment]
1. Retry service in RPC to be able to debounce pending txs in mongo and shoot them again 
2. Prune pending txs when they are mined (update some "mined" field or delete it, tbd)
3. Decide on max retry
4. Limit invalid txs as early as possible in send_raw_transaction, e.g. check chain_id, only check synchronous attributes (not nonce, not balance, etc.)
